vaadin-select-flow/src/main/java/com/vaadin/flow/component/select/data/SelectDataView.java

```diff
@@ -0,0 +1,51 @@
+package com.vaadin.flow.component.select.data;
+
+import java.util.Optional;
+
+import com.vaadin.flow.data.provider.DataView;
+
+/**
+ * An extended {@link DataView} interface for Select component
+ * 
+ * @param <T>
+ *            item type
+ */
+public interface SelectDataView<T> extends DataView<T> {
```

Comment:
Non blocking side comment - these might be generic for a "SingleSelectDataView" when using in-memory data, but lets not go there at least yet ...
Actually since select doesn't support lazy loading and always fetches everything from the data provider, it might be even possible to include the API directly into the component.